Tribler/Core/Modules/restapi/metadata_endpoint.py

```diff
@@ -140,13 +141,34 @@ def render_POST(self, request):
 
         to_subscribe = bool(int(parameters['subscribe'][0]))
         with db_session:
-            channel = self.session.lm.mds.ChannelMetadata.get(public_key=database_blob(self.channel_pk))
+            channel = self.session.lm.mds.ChannelMetadata.get_for_update(public_key=database_blob(self.channel_pk))
             if not channel:
                 request.setResponseCode(http.NOT_FOUND)
                 return json.dumps({"error": "this channel cannot be found"})
 
             channel.subscribed = to_subscribe
-            channel.local_version = 0
+
+        def delete_channel():
+            # TODO: this should be eventually moved to a garbage-collector like subprocess in MetadataStore
+            with db_session:
+                c = self.session.lm.mds.ChannelMetadata.get_for_update(public_key=database_blob(self.channel_pk))
```

Comment:
Minor comment: I always try to avoid one letter variable names. Could you rename it to `channel` for instance?

Reply:
Long and descriptive variable names should only be used for long-lived variables. If a variable lives only a few lines of code, and every usage fits in a single screen, it is perfectly fine to use very short, even single-character name. It makes lines shorter and more readable. The programmer "loads" the variable meaning in his brain, and it persists there for a few lines of code.

In fact, using single-character names even at long spans of code is fine if it fits one of two special use cases:
1) There is an unspoken arrangement that some letter stands for a variable with a special usage type, e.g. `I` and `j` stand for a loop counter, `g` stands for `lambda` in Pony expressions, etc;
2) It stands for a very often used pointer, like `self` (mostly used in C).

Reply:
In this situation, the fact that you are using a one letter variable name induced additional time to understand what's going on. When reviewing the code, I did not understand what the type of `c` was (also complicated by the fact that Python is essentially typeless), so I had to read the rest of the line in more detail. If the variable would be named `channel`, it would be immediately clear for me and it would likely avoid the (minor) overhead to defer the type of `c`.

> It makes lines shorter

While it makes lines shorter, I prefer clearness above shorter and more compact code. Since we do not have type information and annotations, naming variables appropriately in Python is important.

Reply:
You're right.
```
channel c = self.session.lm.mds.ChannelMetadata.get_for_update(public_key=database_blob(self.channel_pk))```
would provide a mental shortcut, however
```
c = self.session.lm.mds.ChannelMetadata.get_for_update(public_key=database_blob(self.channel_pk)) ```
requires the programmer to read the whole line, which is inconvenient.
Unfortunately, Python is typeless.
For me, combination of  `ChannelMedatata` with `get` unambigously point to the fact that `c` is of ChannelMetadata type. But that is because I am familiar with Pony, and I use a consistent notation for one-letter identifiers.
The fact that it took you some additional effort to derive the data type tells us that for an average Python programmer Pony notation does not indicate the type clearly enough.

I will change the identifier.